### ISSUE TITLE
Faster count

### ISSUE THREAD

**Issue by PallHaraldsson** [JuliaData/DataFrames.jl#2849]
Hi, @bkamins, since I like your https://github.com/JuliaLang/julia/pull/40564 code (with the "forgetmenot" label, apparently forgotten) so much, wanted it in 1.7 (and 1.6) but will not happen, I was thinking is there a workaround.

You could (conditionally) overwrite the function in Base(?). Do you already? I just know of Compat.jl, for new functions, maybe it's also used for changed functions? And optimizations? Or would the overwriting belong here?

**Comment by bkamins** [comment]
Things in Julia Base are usually slow. I think Compat.jl is not a place for such changes. But clearly you can add:
```
Base._simple_count(pred, itr, init) = _simple_count_helper(Generator(pred, itr), init)

function _simple_count_helper(g, init::T) where {T}
...
```

code to your projects and unless you are not exporting them to the public this should not be that bad.

I am closing the issue here as it is not DataFrames.jl related. I propose that if you want to keep the discussion (and maybe get an opinion of other people) to post on Julia Discourse.

Thank you!

**Comment by PallHaraldsson** [comment]
At least your PR got merged. I not your "This doesn't affect DataFrames.jl directly", but it does indirectly and DF would be blamed for slowness? It's regrettable it just wasn't merged earlier for 1.7:

https://github.com/JuliaLang/julia/pull/40564#issuecomment-883088597

but we couldn't have known back in July 1.7 would be this delayed. Do you have any idea of the regression potential if backported to 1.7? Or policy for, maybe ok for 1.7.1?

Same for new functions, seems like no risk for 1.7:
https://github.com/JuliaLang/julia/pull/40995

**Comment by bkamins** [comment]
The usual way is to comment in the PR in JuliaLang/julia asking for backport and core devs will decide if it is feasible.
